pkg/workloadmeta/collectors/internal/docker/image_sbom_trivy.go

```diff
@@ -81,22 +82,30 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 
 	go func() {
 		for result := range resultChan {
+			status := workloadmeta.Success
+			reportedError := ""
+			var report *cyclonedx.BOM
 			if result.Error != nil {
 				// TODO: add a retry mechanism for retryable errors
 				log.Errorf("Failed to generate SBOM for docker: %s", result.Error)
-				continue
-			}
-
-			bom, err := result.Report.ToCycloneDX()
-			if err != nil {
-				log.Errorf("Failed to extract SBOM from report")
-				continue
+				status = workloadmeta.Failed
+				reportedError = result.Error.Error()
+			} else {
+				bom, err := result.Report.ToCycloneDX()
+				if err != nil {
+					log.Errorf("Failed to extract SBOM from report")
+					status = workloadmeta.Failed
+					reportedError = result.Error.Error()
+				}
+				report = bom
```

Comment:
imo this code can be refactor with what was changed for containerd.

Reply:
I do agree. The entire code could be refactored with generics here